### PR TITLE
Cloud store tables

### LINKED ISSUE #21: Refactor database Tables for optimal performance
Consider creating a Locations table since that's the piece of data that will be transferred frequently.

Groups should have only Name and list of  User Ids

Users should have profile data, no  location data which is constantly changing

### PR DESCRIPTION
**Pull request by alankley**
Lots of changes in this PR.

- Updated libraries to latest version.
- Created new CloudStore tables (https://console.firebase.google.com/u/0/project/trovami-bcd81/firestore/data~2Fgroups~2Ftest)
- Created CloudFirebaseHelper to manage communication with Cloud Firestore for most tables (locations the exception)
- Created Group2 & Group2Manager until we're ready to deprecate existing Group.  When we update other screens using existing Group & Firestore api calls, we can deprecate Group and rename Group2 as Group
- Added UnitTests route accessible from Groups Route by selecting Pencil Icon
- Introduced ThemeManager to make it easier to use set of Colors/Styles for app. This can easily be expanded to use more than one theme for Night Mode for example.  In my Sandy Scores app, I actually called it a "Sunny" theme that used white on black which gave a better contrast outside

### CODE REVIEW

File: lib/helpers/CloudFirebaseHelper.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+//import 'ItemsHelper.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:trovami/model/DocItem.dart';
+
+class CloudFirebaseHelper {
+  Future addItemToCloud(DocItem item, String collectionName) async{
```
Comment (Samaritan1011001) [inline lib/helpers/CloudFirebaseHelper.dart]: It's a small thing but maybe we can call these function names a bit more simpler? How about "postItem"

File: lib/helpers/CloudFirebaseHelper.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  Future deleteItem(DocItem item, String collectionName) async {
+    CollectionReference collection = FirebaseFirestore.instance.collection(collectionName);
+
+    collection.doc(item.id)
+        .delete()
+        .then((value) => print("FirebaseHelper.deleteItem(): ${item.getName()} deleted"))
+        .catchError((error) => print("Failed to delete ${item.getName()}: $error"));
+  }
+
+  Future<Map<String, Object>> fetchDocs(String collectionName, DocItem item) async{
```
Comment (Samaritan1011001) [inline lib/helpers/CloudFirebaseHelper.dart]: "getItem" ?

File: lib/helpers/CloudFirebaseHelper.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      .get()
+      .then((QuerySnapshot querySnapshot) => {
+        querySnapshot.docs.forEach((doc) {
+          items.putIfAbsent(doc.id, () => item.fromData(doc.data()));
+        })
+      });
+    return items;
+  }
+
+  Future updateToCloud(DocItem item, collectionName) async{
```
Comment (Samaritan1011001) [inline lib/helpers/CloudFirebaseHelper.dart]: "updateItem"

File: lib/helpers/CloudFirebaseHelper.dart (hunk 1 above)
Comment (alankley) [reply lib/helpers/CloudFirebaseHelper.dart]: Naming is very important to me.  I like your suggestions.
I like to use different verbs for retrieving data from cloud, vs retrieving data from local source.
Post is clearly to the cloud, "Save" or something similar we could use to store locally.
I like to us Fetch for the cloud and Get locally. But I'll go with Get for cloud now.

File: lib/helpers/CloudFirebaseHelper.dart (hunk 2 above)
Comment (alankley) [reply lib/helpers/CloudFirebaseHelper.dart]: I went with getItems for now.  What would the function be to retrieve from local storage?
If we need to "Get" a particular User or Group from local storage, seeing "getItem" would be unclear if we are hitting the online database or retrieving it locally.  Of course the Class being used would document the source but sometimes the class is not included in the call.

File: lib/helpers/CloudFirebaseHelper.dart (hunk 3 above)
Comment (alankley) [reply lib/helpers/CloudFirebaseHelper.dart]: yes

File: lib/helpers/CloudFirebaseHelper.dart (hunk 1 above)
Comment (Samaritan1011001) [reply lib/helpers/CloudFirebaseHelper.dart]: Yeah that's correct but we use `CloudFirebaseHelper().postItem()` right? so the object `CloudFirebaseHelper()` itself tells it is to the cloud. It its a class that store locally we would call it `LocalStoragehelper()` so both cases the object conveys where it's going and the method call tells which action.

File: lib/helpers/CloudFirebaseHelper.dart (hunk 2 above)
Comment (Samaritan1011001) [reply lib/helpers/CloudFirebaseHelper.dart]: Yeah, I don't mind the current name either.

**Comment by Samaritan1011001** [comment]
LGTM

**Comment by Samaritan1011001** [comment]
I'll merge once you have seen the comments and give a thumbs up on no more changes for this PR

**Pull request merged**